dataflow/src/org/checkerframework/dataflow/analysis/FlowExpressions.java

```diff
@@ -407,7 +407,8 @@ private static Receiver internalRepOfMemberSelect(
             case ENUM:
             case INTERFACE: // o instanceof MyClass.InnerInterface
             case ANNOTATION_TYPE:
-                return new ClassName(expressionType);
+                TypeMirror selectType = TreeUtils.typeOf(memberSelectTree);
```

Comment:
The motivation for this change is that previously, for a fully-qualified class name like `p.q.C` it used `p.q` as `ClassName`, which doesn't make sense. But maybe I misunderstood and this needs documentation.